### PR TITLE
FUTURE: support `deno install <alias>@npm:<package>`

### LINKED ISSUE #23144: Support `deno add` with aliases
For example, npm is:

```sh
npm install --save-dev typescript-3.6.3@npm:typescript@3.6.3 typescript-4.0.2@npm:typescript@4.0.2
```

We should probably have the exact same syntax:

```sh
deno add typescript-3.6.3@npm:typescript@3.6.3 typescript-4.0.2@npm:typescript@4.0.2
```

Or with https urls:

```sh
deno add path@https://deno.land/std/path/mod.ts fs@https://deno.land/std/fs/mod.ts
```

### PR DESCRIPTION
**Pull request by dsherret**
Closes #23144

### CODE REVIEW

File: cli/tools/registry/pm.rs [hunk 1]
```diff
@@ -456,3 +513,53 @@ fn update_config_file_content(
...
+        value: AddPackageReqValue::Npm(PackageReq::from_str("foo").unwrap())
+      }
+    );
+    assert_eq!(
+      AddPackageReq::parse("@alias/pkg@jsr:foo").unwrap(),
+      AddPackageReq {
+        alias: "@alias/pkg".to_string(),
+        value: AddPackageReqValue::Jsr(PackageReq::from_str("foo").unwrap())
+      }
+    );
```
Comment (bartlomieju) [inline cli/tools/registry/pm.rs:552]: What about `alias@jsr:foo@^1.5.0`? Is it handled here? If not can you add a test that expects it to fail?

**Review by bartlomieju — APPROVED** [review]

File: cli/tools/registry/pm.rs (hunk 1 above)
Comment (dsherret) [reply cli/tools/registry/pm.rs:552]: That will succeed and then fail on download. I added a test for it succeeding.

Comment (bartlomieju) [reply cli/tools/registry/pm.rs:552]: Why will it fail on download?

Comment (dsherret) [reply cli/tools/registry/pm.rs:552]: Because it's not a scoped package.

**Pull request merged**